File: src/backend/main/java/com/retrobased/market/dto/OrderDTO.java
package com.retrobased.market.dto;

import com.fasterxml.jackson.annotation.JsonInclude;

import java.time.OffsetDateTime;
import java.util.UUID;

public record OrderDTO(
        UUID orderId,

        CustomerAddressDTO address,

        OffsetDateTime orderApprovedAt,

        @JsonInclude(JsonInclude.Include.NON_NULL)
        OffsetDateTime orderDeliveredCarrierDate,

        @JsonInclude(JsonInclude.Include.NON_NULL)
        OffsetDateTime orderDeliveredCustomerDate,

        OffsetDateTime createdAt
) {
}

File: Frontend/src/app/services/models/order-dto.ts
export interface OrderDTO {
    id: string;
    address: {
      addressLine1: string;
      city: string;
    };
    createdAt: string;
    orderApprovedAt: string;
    orderDeliveredCarrierDate?: string;
    orderDeliveredCustomerDate?: string;
  }
  